plugins/nodes/src/js/components/NodesTable-new.tsx

```diff
@@ -95,28 +97,30 @@ export default class NodesTable extends React.Component<
       regionRenderer(this.props.masterRegion, data);
   }
 
-  retrieveSortFunction(sortColumn: string): SortFunction<Node> {
+  retrieveComparators(sortColumn: string): Array<Comparator<Node>> {
+    const equalComparator = () => 0;
+
     switch (sortColumn) {
       case "hostname":
-        return hostnameSorter;
+        return hostnameComparators;
       case "region":
-        return regionSorter;
+        return regionComparators;
       case "zone":
-        return zoneSorter;
+        return zoneComparators;
       case "health":
-        return healthSorter;
+        return healthComparators;
       case "tasks":
-        return tasksSorter;
+        return tasksComparators;
       case "cpu":
-        return cpuSorter;
+        return cpuComparators;
       case "mem":
-        return memSorter;
+        return memComparators;
       case "disk":
-        return diskSorter;
+        return diskComparators;
       case "gpu":
-        return gpuSorter;
+        return gpuComparators;
       default:
-        return (data, _sortDirection) => data;
+        return [equalComparator];
```

Comment:
Could it be a potential source of confusion that whenever you have a single comparator you have to wrap it on a list? 

Can we now, or on future work, provide an API that can detect a single function and wrap it on the array for you?

Reply:
I originally wanted to use a spread to do the trick, but I found the API to be kinda convoluted then. Would you prefer something like this instead?

```js
sort(comparatorA, comparatorB, comparatorC, data) // Option A
sort(comparatorA, comparatorB, comparatorC)(data) // Option B
```

On second thought, in this case here we could also write an empty array, would  you like this better?

Reply:
I think my ideas was more like, you cam pass a Function or a list of Function and the API does what you would expect. Just to solve for the single function case. 

For many comparators, I am fine with that being a list.

Reply:
I don't feel really good about this. You trade `[]` on the call side with

Reply:
But don't you think as a package maintainer it would be great to make the best api for whoever uses your lib? I really feel like caring about creating a list of one makes the dev-user live harder.

What about having an union type? Would that give you the type guarantees you want `type comparator = Comparator<T> | Array<Comparator<T>>);`. We could combine that with a user-defined-type-guard to get the behavior of  `makeArrayIfItIsntAlready` bad at all. 

You could maybe do all of this by just introducing a nother function who offers this api, and smartly call the underlying sort with the Array<Comparator<T>>.

Reply:
I second @Fabs here, having the possibility to call `sort(comparator, data)` could be beneficial🤔 

assuming my change above, we could do something like this:

```
export function sort<T>(comparators: Array<Comparator<T>> | Comparator<T>, data: T[]): T[] {
  return (Array.isArray(comparators)) 
    ? [...data].sort((a,b) => sortByComparators(a, b, comparators)) 
    : [...data].sort(comparator);
}
```
we also could export two functions if you want to keep the interface strict: `sortByComparator` and `sortByComparators`:
```
export function sortByComparator<T>(comparator: Comparator<T>, data: T[]): T[] {
  return [...data].sort(comparator);
}
export function sortByComparators<T>(comparators: Array<Comparator<T>>, data: T[]): T[] {
  return [...data].sort((a,b) => sortByComparators(a, b, comparators))
}
```
(again, these are assuming my change from above)

Reply:
> But don't you think as a package maintainer it would be great to make the best api for whoever uses your lib?

Reply:
> But don't you think as a package maintainer it would be great to make the best api for whoever uses your lib? 

Yes I do think so, but the best API IMHO is not the one with the most indirection, but with the clearest and most concise API. I am aware that my opinion here is pretty dogmatic, but I don't see why cases like this should be handled here. If you argue I would like to be able to call it with only a comparator then we could also argue we want it to be able to handle null or undefined, because I don't want to check if my variable exists first. 

I chose this trade-off because I wanted to have the simplest API possible, which does not need a lot of documentation to understand it. Just give it a list of comparators and that's it.